.github/workflows/build-and-test-scheduled.yaml

```diff
@@ -35,3 +35,10 @@ jobs:
           rosdistro: galactic
           target-packages: ${{ steps.get-self-packages.outputs.self-packages }}
           build-depends-repos: build_depends.repos
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          files: lcov/total_coverage.info,coveragepy/.coverage
+          fail_ci_if_error: true
+          verbose: true
```

Comment:
Where can we check this result?
Since it's not a PR workflow but a scheduled workflow, I guess it won't comment on PRs.

Reply:
Yes, I am not going to comment on the PR. I added this step in order to send the overall coverage results to CodeCov.

However, I realized that I need to add a new workflow to the `tier4/main` branch to run build-and-test job after merging the PR. And instead of a scheduled job, we need to add a CodeCov step to it.
This will enable us to calculate the overall coverage of `tier4/main` and enable CodeCov to correctly detect the coverage difference in PR.

Here is an example warning for not sending the results of the overall coverage of the base branch to CodeCov.
![image](https://user-images.githubusercontent.com/19993104/150897607-d5715c76-f56f-407f-9484-bdd36ba7a897.png)

https://github.com/KeisukeShima/autoware.universe/pull/2#issuecomment-1019943871

May I create a new PR in order to add a workflow after merging PRs?

Reply:
Yes, please!

Reply:
@KeisukeShima BTW, what is `a new workflow`? :thinking: Could you explain it a bit more in detail?

Reply:
New workflow is same as `build-and-test-scheduled` job, but it will be executed after merging the PRs.

Reply:
Why scheduled workflow isn't acceptable?

Reply:
CodeCov calculates the diff coverage using the coverage results calculated on a pull request base.
In this case, a scheduled workflow is not sufficient because it is only executed once a day. In other words, the workflow will not be executed for all commits in the `tier4/main` branch.
Therefore, CodeCov is not getting coverage results for each commit in the `tier4/main` branch. This means that the coverage results calculated on a pull request base may not be found and CodeCov may fail.

See comment below. (Note that the comment specifies the commit sha as well as the branch).
> No coverage uploaded for pull request base (**tier4/proposal@10bd1c0**)
https://github.com/autowarefoundation/autoware.universe/pull/292#issuecomment-1020797172

Reply:
Also, I think it is important to run the `build-and-test` workflow when the PR is merged. It can detect any side effects caused by the added/removed code.

Reply:
@KeisukeShima Do you mean `build-and-test-pr` isn't enough? Could you please explain the reason?

Reply:
Yes. There are some reasons:
* `build-and-test` can detect any side effects caused by the added/removed code from PRs.
* Since `build-and-test-scheduled` is executed on a daily basis, if there are several PRs in a day and `build-and-test-scheduled` fails, we need to investigate all the PRs that were merged the day before. However, if `build-and-test` fails, we only need to investigate a single PR.
* The build-and-test-scheduled is still necessary because it can detect errors that occur even if there are no changes in the code. For example, it is caused by the upgrading of third party libraries.

Reply:
Okay, thank you!